### PR TITLE
Move the construction of QueryContext inside Foreman.

### PR DESCRIPTION
**Pull request by zuyu**
This PR moves the construction of `QueryContext` inside `Foreman`. And for the distributed case, inside `Shiftboss`.

This may affect @hbdeshmukh 's work on learning-based scheduling mechanism, particular `QueryManager`.

### CODE REVIEW

File: query_execution/Foreman.hpp [hunk 1]
```diff
@@ -522,6 +525,7 @@ class Foreman : public Thread {
   WorkerDirectory *workers_;
 
   friend class ForemanTest;
+  FRIEND_TEST(ForemanTest, TwoNodesDAGPartiallyFilledBlocksTest);
```
Comment (hbdeshmukh) [inline query_execution/Foreman.hpp]: Why is this line needed?

Comment (zuyu) [inline query_execution/Foreman.hpp]: `TwoNodesDAGPartiallyFilledBlocksTest` accesses `Foreman`'s private member `query_context_`  to mock to add partially filled blocks in `InsertDestination`.

**Comment by hbdeshmukh** [comment]
Looks good to me. Merging.
